src/ext-test/java/org/opentripplanner/ext/siri/SiriTimetableSnapshotSourceTest.java

```diff
@@ -565,6 +549,80 @@ public TripTimes getTripTimesForTrip(Trip trip) {
       return getTripTimesForTrip(trip.getId(), serviceDate);
     }
 
+    public String getRealtimeTimetable(String tripId) {
+      return getRealtimeTimetable(id(tripId), serviceDate);
+    }
+
+    public String getRealtimeTimetable(Trip trip) {
+      return getRealtimeTimetable(trip.getId(), serviceDate);
+    }
+
+    public String getRealtimeTimetable(FeedScopedId tripId, LocalDate serviceDate) {
+      var tt = getTripTimesForTrip(tripId, serviceDate);
+      var pattern = getPatternForTrip(tripId);
+
+      return encodeTimetable(tt, pattern);
+    }
+
+    public String getScheduledTimetable(String tripId) {
+      return getScheduledTimetable(id(tripId));
+    }
+
+    public String getScheduledTimetable(FeedScopedId tripId) {
+      var pattern = getPatternForTrip(tripId);
+      var tt = pattern.getScheduledTimetable().getTripTimes(tripId);
+
+      return encodeTimetable(tt, pattern);
+    }
+
+    /**
+     * This encodes the times and information about stops in a readable way in order to simplify
+     * testing. The format is:
+     *
+     * <pre>
+     * REALTIME_STATE | stop1 [FLAGS] arrivalTime departureTime | stop2 ...
+     *
+     * Where flags are:
+     * C: Canceled
+     * R: Recorded
+     * PI: Prediction Inaccurate
+     * ND: No Data
+     * </pre>
+     */
+    private String encodeTimetable(TripTimes tripTimes, TripPattern pattern) {
```

Comment:
I like this, maybe it can be moved into the core code with TripTimes and TripPattern. For example in a utility class like `TripTimesStringBuilder`  (like the PathStringBuilder) - then we can reuse it across tests an if we want also use i in debugging/logging. If you want you can put it in the test source tree, and we can move it to main if needed later. I think it is usful in both test and main - it is up to you.

Reply:
Good idea! I moved it into main.